messenger/http_transporter.go

```diff
@@ -490,7 +490,10 @@ func (t *HTTPTransporter) messageDecoder(w http.ResponseWriter, r *http.Request)
 	// libprocess will ignore responses, see ignore_data).
 	// see https://github.com/apache/mesos/blob/adecbfa6a216815bd7dc7d26e721c4c87e465c30/3rdparty/libprocess/src/process.cpp#L2192
 	for {
-		if r, ok := <-decoder.Requests(); ok {
+		if r, ok := <-decoder.Requests(); !ok {
```

Comment:
This was the problem - decoder.Requests() closes, this turns into a tight loop that just keeps trying to read from the already closed channel.  The solution is to exit the loop once the channel is closed.